File: plugins/devkit/devkit-java-tests/testData/inspections/internal/com/intellij/openapi/extensions/PluginId.java
package com.intellij.openapi.extensions;

public final class PluginId {

  void test() {
    assert this == this;
    assert this != this;

    PluginId first = new PluginId();
    PluginId second = new PluginId();

    assert first == null;
    assert first != null;

    assert this == second;
    assert second != this;

    assert <warning descr="'PluginId' instances should be compared for equality, not identity">first == second</warning>;
    assert <warning descr="'PluginId' instances should be compared for equality, not identity">first != second</warning>;

    assert <warning descr="'PluginId' instances should be compared for equality, not identity">second == first</warning>;
    assert <warning descr="'PluginId' instances should be compared for equality, not identity">second != first</warning>;
  }
}

File: plugins/devkit/devkit-kotlin-tests/testData/inspections/internal/PluginId.kt
package com.intellij.openapi.extensions

class PluginId {

  fun test() {
    if (this === this) {}
    if (this !== this) {}

    val first: PluginId = PluginId()
    val second: PluginId? = first

    if (this === second) {}
    if (second !== this) {}

    if (<warning descr="'PluginId' instances should be compared for equality, not identity">first === second</warning>) {}
    if (<warning descr="'PluginId' instances should be compared for equality, not identity">first !== second</warning>) {}

    if (<warning descr="'PluginId' instances should be compared for equality, not identity">second === first</warning>) {}
    if (<warning descr="'PluginId' instances should be compared for equality, not identity">second !== first</warning>) {}

    val third: MyPluginId = first

    if (<warning descr="'PluginId' instances should be compared for equality, not identity">third === first</warning>) {}
    if (<warning descr="'PluginId' instances should be compared for equality, not identity">third !== first</warning>) {}
  }
}

typealias MyPluginId = PluginId